[Preprocessing] Fix bug in TD dag matching op

Max191 opened the pull request:
Fixes a bug in the `transform.iree.match.cast_compatible_dag_from_root` op when there are repeated operands.

hanhanW committed:
Bump llvm to llvm/llvm-project@c06d0ff806b7 (#19903)

kuhar committed:
[Codegen][LLVMGPU] Add pass to materialize pad encoding (#19909)

This is based on the materialize-encoding-into-nop pass, with additional
patterns to handle load and store ops.
For now, we materialize load of a padded input with and extract slice
and store as insert slice into a larger tensor. These get folded and
become partial loads/stores at the end, but we can change this later.

Enable this by default in the LLVMGPU pass pipeline as it's a superset
of the existing nop encoding materialization pass.

---------

Signed-off-by: Jakub Kuderski <jakub@nod-labs.com>

Commit:
[runtime][python] add f8 element types (#19928)

These types are not available in NumPy so no interoperability is
provided for them.

---------

Signed-off-by: Boian Petkantchin <boian.petkantchin@amd.com>

hanhanW committed:
[Integrate] Add arg/res_attrs to ops that implement CallOpInterface. (#19926)

It drops the revert of
https://github.com/llvm/llvm-project/commit/327d627066e6452b081365b595657d17f2690a3b

---------

Signed-off-by: hanhanW <hanhan0912@gmail.com>

ScottTodd committed:
Refresh compile/run examples in deployment configuration guides. (#19920)

Progress on https://github.com/iree-org/iree/issues/18174, updating some
stale documentation.

> [!NOTE]
> Demo here:
https://scotttodd.github.io/iree/guides/deployment-configurations/cpu/

Changes included:

* Switch examples to use ONNX instead of TensorFlow given that users are
trying to use TensorFlow and failing:
https://github.com/iree-org/iree/issues/19852
* Add more documentation for CPU targets and features for
https://github.com/iree-org/iree/issues/18561
* Standardize some formatting across CPU/CUDA/ROCm/Vulkan pages
* Adjust some parts of the ONNX guide now that support is more mature

hanhanW committed:
[CUDA][Integrate] Switch CUDATarget to ptx_kernel cc. (#19925)

This drops the local LLVM revert:
https://github.com/llvm/llvm-project/commit/de7438e472839df63e1f10478436c2f15b8e4184

Signed-off-by: hanhanW <hanhan0912@gmail.com>

hanhanW committed:
[Stream] Add layouts to encodings for all stream tensor AffinityOp. (#19726)

The revision adds the support for the rest of AffinityOp that have
TensorPhase trait, i.e., TensorCloneOp, TensorSliceOp, TensorFillOp, and
TensorUpdateOp ops. It is tricky to handle encodings for transfer ops,
so only the encoding in the fill op is updated. If other operations have
tensor encodings, it returns a failure for now.

There are two stream tensor ops do not implement the
AffinityOpInterface, so they are not supported within the revision. They
are stream.tensor.load op and stream.tensor.store op. We should be able
to track the resource affinity for these two ops, and it requires
additional analysis. Thus, they are not scoped within the revision.

The revision also adds the missing documentation to the
`addLayoutsToTensorPhaseOps` method.

---------

Signed-off-by: hanhanW <hanhan0912@gmail.com>

hanhanW committed:
Integrate llvm to llvm/llvm-project@e470dcae8d2c41 (#19927)

Revert commits:

-
https://github.com/llvm/llvm-project/commit/8c1dbac304d2f3d9e9685c1f4fd8b3e133360893

The author is working on a fix, and it is not ready yet.

---------

Signed-off-by: hanhanW <hanhan0912@gmail.com>

hanhanW committed:
Integrate llvm to llvm/llvm-project@a1984ec5eab09f (#19934)

Signed-off-by: hanhanW <hanhan0912@gmail.com>

Max191 committed:
[Preprocessing] Fix bug in TD dag matching op

Signed-off-by: Max Dawkins <max.dawkins@gmail.com>

MaheshRavishankar reviewed (approved):
Stamped on condition that tests will be added subsequently.

qedawkins committed:
[Codegen][GPU] Also don't tile large fills (#19937)

This skips tiling large fills for the same reasoning as in #19887

bjacob committed:
LLVM: cherry-pick 73f11ac (#19939)

Cherry pick
https://github.com/llvm/llvm-project/commit/73f11ac17d50f0585d03a880e756ff13c321bfd0

Per @ScottTodd : "fixes Windows builds + our nightly releases".


https://discord.com/channels/689900678990135345/1080178290188374049/1337527415492182137

Signed-off-by: Benoit Jacob <jacob.benoit.1@gmail.com>

bjacob committed:
Integrate LLVM at 73f11ac (#19941)

We had previously cherry-picked
https://github.com/llvm/llvm-project/commit/73f11ac17d50f0585d03a880e756ff13c321bfd0
in https://github.com/iree-org/iree/pull/19939.

Now we're integrating up to that commit, so it's no longer a
cherry-pick.

Reverting https://github.com/llvm/llvm-project/pull/125789 because it
breaks TorchToTosa, in torch-mlir. We will need to wait for this to be
resolved in torch-mlir, then simultaneously bump torch-mlir and drop the
revert.

Chery-pick a Bazel fix:
https://github.com/llvm/llvm-project/commit/4df287a171b63514a5028b85272fcc1b89555ee4

---------

Signed-off-by: Benoit Jacob <jacob.benoit.1@gmail.com>

Max191 committed:
add test for matcher

Signed-off-by: Max Dawkins <max.dawkins@gmail.com>
